Implement stop all actions

jahazielhigareda opened the pull request.

jahazielhigareda committed:
233 - Implement stop all actions

jahazielhigareda committed:
233 - Implement stop all actions

jahazielhigareda committed:
233 - Remove StopAllActionsCommand and add StopAllActions into player.cs

caioavidal commented on src/Game/NeoServer.Game.Creatures/Model/Players/Player.cs:
```diff
@@ -917,6 +917,21 @@ public override ILoot DropLoot()
 
+    public void StopAllAction()
+    {
+        if (Attacking)
```
I'd remove those ifs and call the 3 stops directly
StopAttack();
StopWalking();
StopFollowing();

jahazielhigareda committed:
290 - Remove CancelWalk and substitute for StopWalk and add validate if have next step

jahazielhigareda committed:
Fix uTest WalkTo_Should_Emit_Events_And_Add_Next_Steps

jahazielhigareda committed:
Fix builder map build

jahazielhigareda committed:
Fix UTest - Stop_following_interrupts_player_walk

jahazielhigareda committed:
Create UTest
StopAllActions when Attacking
StopAllActions when Following
StopAllActions when Walking

caioavidal commented on src/Game/NeoServer.Game.Common/Contracts/Creatures/IPlayer.cs:
```diff
@@ -216,4 +216,5 @@ public interface IPlayer : ICombatActor, ISociableCreature
     void Read(IReadable readable);
     event ReadText OnReadText;
     void Write(IReadable readable, string text);
+    void StopAllAction();
```
Can you rename to "StopAllActions()" ?

caioavidal commented on tests/NeoServer.Game.Creatures.Tests/WalkableCreature/PlayerTest.cs:
```diff
@@ -296,4 +299,136 @@ public void PlayerDoesNotGainSkillsWhenUsingAHealingRune()
+
+        sut.SetCurrentTile(tile.Object);
+        //First instruction walking
+        sut.WalkTo(new Location(105, 100, 7));
```
He we don't need WalkTo and instead Attack

jahazielhigareda replied:
This method is not use, i will remove

jahazielhigareda committed:
Remove dont use method

jahazielhigareda committed:
Change name StopAllAction to StopAllActions

caioavidal commented:
Looks good. thanks

caioavidal commented:
close #233
